corehq/apps/hqadmin/views.py

```diff
@@ -841,6 +842,44 @@ def _malt_csv_response(month, year):
     return export_as_csv_action(exclude=['id'])(MALTRowAdmin, None, queryset)
 
 
+class DownloadGIRView(BaseAdminSectionView):
+    urlname = 'download_gir'
+    page_title = ugettext_lazy("Download GIR")
+    template_name = "hqadmin/gir_downloader.html"
+
+    @method_decorator(require_superuser)
+    @use_bootstrap3
+    def dispatch(self, request, *args, **kwargs):
+        return super(DownloadGIRView, self).dispatch(request, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        from django.core.exceptions import ValidationError
+        if 'year_month' in request.GET:
+            try:
+                year, month = request.GET['year_month'].split('-')
+                year, month = int(year), int(month)
+                return _gir_csv_response(month, year)
+            except (ValueError, ValidationError):
+                messages.error(
+                    request,
+                    _("Enter a valid year-month. e.g. 2015-09 (for December 2015)")
+                )
+        return super(DownloadGIRView, self).get(request, *args, **kwargs)
+
+
+def _gir_csv_response(month, year):
+    query_month = "{year}-{month}-01".format(year=year, month=month)
+    queryset = GIRRow.objects.filter(month=query_month)
+    field_names = GIR_FIELDS
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = u'attachment; filename=gir.csv'
+    writer = csv.writer(response)
+    writer.writerow(list(field_names))
+    for obj in queryset:
+        writer.writerow(obj.export_row)
+    return response
```

Comment:
the csv generator used by the malt report does not work in this case, because the columns for the GIR report must be in a specific order and dont map one to one with the fields of the model